PLAT-9983: Migrate Poll Bot to BDK2.0

symphony-hong opened the pull request:
Migrate Poll Bot from using Java Legacy SDK to BDK2.0.
Migration guide: https://perzoinc.atlassian.net/wiki/spaces/DevX/pages/1492814947/PLAT-9983+Migration+guide+for+Poll+bot+from+Java+SDK+to+BDK2.0

symphony-hong committed:
Migrate to BDK2.0

symphony-youri commented on pom.xml:
```diff
@@ -31,42 +31,67 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-data-mongodb</artifactId>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/com.symphony.platformsolutions/symphony-bdk-template-freemarker -->
```
do we need the comment? and below too

symphony-youri commented on src/main/java/com/symphony/ps/pollbot/utils/MessageUtils.java:
```diff
@@ -0,0 +1,67 @@
+        tokens.put("\\{", "&#123;");
+        tokens.put("\\}", "&#125;");
+
+        pattern = Pattern.compile("(" + StringUtils.join(tokens.keySet(), "|") + ")");
```
we could get it of apache commons with String.join

symphony-youri commented on src/main/resources/application.properties:
```diff
@@ -7,3 +7,7 @@ logging.level.org.mongodb.driver.connection=error
 spring.config.location=file:./src/main/resources/application.properties
 spring.data.mongodb.uri=
+
+bdk.host=acme.symphony.com
```
we need to migrate the config.json here (and drop the config.json file)

symphony-youri replied:
and we also need to figure out with @ystan- how it is configured when deployed for real

symphony-hong replied:
I'm not sure about the RSA key path. @ystan- Could you clarify the config of the poll bot please?

symphony-youri commented on src/main/java/com/symphony/ps/pollbot/RealTimeEventComponent.java:
```diff
@@ -0,0 +1,45 @@
+    public void onMessageSent(RealTimeEvent<V4MessageSent> event) throws PresentationMLParserException {
+        pollService.handleIncomingMessage(
+                event.getSource().getMessage(),
+                StreamType.TypeEnum.fromValue(event.getSource().getMessage().getStream().getStreamType()));
```
if the api changes and a new stream type is added this might break
so we could be careful and handle this case and not try to process the event it has an unknown type

symphony-hong replied:
so I think we should only handle the incoming message from IM and ROOM?

symphony-youri reviewed (commented):
We could have used @slash commands too I guess (but it requires more changes)

symphony-hong committed:
Update Youri's comments

Use String#join instead of StringUtils#join to get rid of commons-lang lib.

Migrate bot configuration from config.json to application.properties.

Well-handle the stream type of the incoming message to be sure that the bot will not down if there's any new stream type in future.

Clean up the pom file.

symphony-youri commented:
Ok for me but I think we should wait for @ystan- feedback on the configuration part

symphony-youri commented:
Fixing Snyk reported issue in the upcoming 2.0.0 version of the BDK
